src/client.rs

```diff
@@ -0,0 +1,45 @@
+use std::net::UdpSocket;
+use std::sync::Arc;
+
+pub trait ClientTrait {
+    fn get_id(&self) -> u16;
+    fn get_id_byte(&self) -> &[u8];
+    fn read_loop(&mut self);
+}
+
+struct Client {
+    id: u16,
+    id_byte: [u8; 2],
+    sock: Arc<UdpSocket>,
+}
+
+impl Client {
+    pub fn new(id: u16, addr: String) -> Result<Self, anyhow::Error> {
+        let sock = UdpSocket::bind(addr);
+        if let Err(e) = sock {
+            return Err(anyhow::anyhow!(e));
+        }
+        let id_byte = id.to_be_bytes();
+        let sock = Arc::new(sock.unwrap());
```

Comment:
`unwrap()` は避けたほうが行儀良いかも

```suggestion
        let sock = UdpSocket::bind(addr)?;
        let id_byte = id.to_be_bytes();
        let sock = Arc::new(sock);
```

Reply:
これって `UdpSocket::bind(addr)?;` でエラーが返されたらどうなるの？

Reply:
ここは

```rs
let sock = match UdpSocket::bind(addr) {
    Ok(s) => s,
    Err(e) => return Err(e.into()),
};
```

これと同じ